genesyscloud/resource_genesyscloud_integration.go

```diff
@@ -0,0 +1,495 @@
+package genesyscloud
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/platformclientv2"
+)
+
+var (
+	integrationConfigResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Integration config name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"notes": {
+				Description: "Integration config notes.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"properties": {
+				Description: "Integration config properties.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"advanced": {
+				Description: "Integration advanced config.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"credentials": {
```

Comment:
Is credentials always a map of string to string? If so, this could be a schema.Map  https://www.terraform.io/docs/extend/schemas/schema-types.html#typemap

Reply:
I used schema.TypeMap at first, then love jsonencode too much and changed it. 
Changed it back and now it's a Map type.

Reply:
These are a map of any credential keys to credential IDs, correct? We should add this to the RefAttrs below so they are excluded from exports until the credentials type is defined. Since the keys are unknown, we can define the attribute as `credentials.*` and it should work as expected.

Reply:
![image](https://user-images.githubusercontent.com/39736275/119558014-e589d600-bd6e-11eb-9872-e2b1904a6c6c.png)
Added reference for integration config credentials.